packages/amplify_auth_cognito/ios/Classes/FlutterSignInRequest.swift

```diff
@@ -23,8 +23,9 @@ struct FlutterSignInRequest {
   init(dict: NSMutableDictionary){
     self.username = dict["username"] as! String?
     self.password = dict["password"] as! String?
-    self.options = dict["options"] as! Dictionary<String, Any>?
+    self.options = dict["options"] as? Dictionary<String, Any>?
```

Comment:
I think we migt need something like this for the options:
```
        let pluginOptions = AWSAuthSignInOptions(
                validationData:   ["somekey": "somevalue"],
                 metadata: ["somekey": "somevalue"]) 
```

Reply:
I kept the same idea from SignUp that build the options on the AuthCognitoBridge.swift

Reply:
And the AuthSignInRequest only has pluginOptions and in my other PR in amplify-ios they confirmed that I must extract the validationData and metadata from pluginOptions.

> We already have a `validationData` on [the plugin options](https://github.com/aws-amplify/amplify-ios/blob/main/AmplifyPlugins/Auth/AWSCognitoAuthPlugin/Models/Options/AWSAuthSignInOptions.swift#L12), we just need to pass it in by extracting it from pluginOptions like we do with [`clientMetadata`](https://github.com/aws-amplify/amplify-ios/pull/1169/files#diff-c3c6164d32c88c5b0ee9c92eb4fec0a8449760dbaf528ae9f6692775685db85aR29), and passing it in on the [AuthenticationProviderAdapter `signIn` call](https://github.com/aws-amplify/amplify-ios/pull/1169/files#diff-c3c6164d32c88c5b0ee9c92eb4fec0a8449760dbaf528ae9f6692775685db85aL33).
> 
> To use it, you'd add your validationData to the `pluginOptions` of the request you create when you sign in.

Reply:
When I made the change of specifically instantiating the `AWSAuthSignInOptions` and passing them in the `pluginOptions` key in `AuthSignInRequest.Options`, I am able to see the values in the Lambda logs.

Reply:
Can you push it so I could test it here? I've had to change the amplify-ios because they was passing nil in validationData and with the actual code from both PR (this one and the one in amplify-ios) the amplify-ios is receiving the Options like this:

```
Options(pluginOptions: Optional(["validationData": {
    alias = 11780859708;
    mobileId = "92939:21sda";
    ofdbChannel = business;
    ofdbLoadError = true;
    sessionId = e1fc3daf909c8afbb8c3889bb082ffa2cc6874bd7a6dee3770e2b54d19a5b335;
}]))
```

But the backend is receiving the userAttributes with validationData and validationData with userAttributes.. like this:

```
userAttributes: {
 'custom:uuid': 'f2805f1c',
 sub: '7c1d0796',
 'cognito:email_alias': 'xxx@btgpactual.com',
 'cognito:user_status': 'CONFIRMED',
 email_verified: 'true',
 'custom:isHubDomain': 'true',
 'custom:origin': 'xxx',
 phone_number_verified: 'false',
 'cognito:mfa_enabled': 'false',
 email: 'xxx@btgpactual.com'
 },
 validationData: {
 'cognito:iOSVersion': '14.5',
 'cognito:bundleShortV': '2.2.01',
 'cognito:bundleVersion': '2.2.11',
 'cognito:idForVendor': 'xxx',
 'cognito:systemName': 'iOS',
 'cognito:bundleId': 'com.package',
 'cognito:deviceName': 'iPhone',
 'cognito:model': 'iPhone'
 },
```

I'm doing something wrong and still couldn't figure it out.

Thanks for your help!

Reply:
I'll do a PR into your branch soon.  Working out a few things.

Reply:
@caravanat I made a PR into your branch.